internal/middleware/middleware_test.go

```diff
@@ -0,0 +1,586 @@
+
+				req := httptest.NewRequest(http.MethodGet, "http://localhost:80", nil)
+				ctx := session.SetSession(req.Context(), model2.SessionStruct{UserId: "123"})
+				mockCacher := redisMock.NewMockCacher(mockCtrl)
+				mockCacher.EXPECT().Get("http://localhost:80/auth/123").Return([]byte("123"), nil)
+				return req.WithContext(ctx), mockCacher
+			},
+			extractMsgFunc: func(closer io.ReadCloser) (string, error) {
+				return "", nil
+			},
+			validator: func(res *httptest.ResponseRecorder) {
+				if hit != false {
+					t.Errorf("Want: %v, Got: %v", false, hit)
+					return
+				}
+				by, _ := ioutil.ReadAll(res.Body)
+				if !reflect.DeepEqual([]byte(""), by) {
+					t.Errorf("Want: %v, Got: %v", "", string(by))
+				}
+			},
+		},
+		{
+			name:   "SUCCESS::Cacher::Normal Response",
+			config: config.Config{Cache: config.CacheCfg{Time: time.Minute}},
+			setupFunc: func() (*http.Request, *redisMock.MockCacher) {
+
+				req := httptest.NewRequest(http.MethodGet, "http://localhost:80", nil)
+				ctx := session.SetSession(req.Context(), model2.SessionStruct{UserId: "123"})
+				mockCacher := redisMock.NewMockCacher(mockCtrl)
+				//x := model.Response{Status: 200, Message: "passed", Data: "123"}
+				//y, _ := json.Marshal(x)
+				//cacheResponse := model2.CacheResponse{Status: http.StatusOK, Response: string(y), ContentType: "application/json"}
+				mockCacher.EXPECT().Get("http://localhost:80/auth/123").Return(nil, errors.New("error"))
+				mockCacher.EXPECT().Set("http://localhost:80/auth/123", gomock.Any(), time.Minute)
+				return req.WithContext(ctx), mockCacher
+			},
+			extractMsgFunc: func(closer io.ReadCloser) (string, error) {
+				return "", nil
+			},
+			validator: func(res *httptest.ResponseRecorder) {
+				if hit != true {
+					t.Errorf("Want: %v, Got: %v", true, hit)
+					return
+				}
+				var resp model.Response
+				by, _ := ioutil.ReadAll(res.Body)
+				json.Unmarshal(by, &resp)
+				expectedResp := model.Response{Status: 200, Message: "passed", Data: "123"}
+				if !reflect.DeepEqual(expectedResp, resp) {
+					t.Errorf("Want: %v, Got: %v", expectedResp, resp)
+				}
+			},
+		},
+		{
+			name:   "Failure::Cacher::Normal Response::Redis fail",
+			config: config.Config{Cache: config.CacheCfg{Time: time.Minute}},
+			setupFunc: func() (*http.Request, *redisMock.MockCacher) {
+
+				req := httptest.NewRequest(http.MethodGet, "http://localhost:80", nil)
+				ctx := session.SetSession(req.Context(), model2.SessionStruct{UserId: "123"})
+				mockCacher := redisMock.NewMockCacher(mockCtrl)
+				//x := model.Response{Status: 200, Message: "passed", Data: "123"}
+				//y, _ := json.Marshal(x)
+				//cacheResponse := model2.CacheResponse{Status: http.StatusOK, Response: string(y), ContentType: "application/json"}
+				mockCacher.EXPECT().Get("http://localhost:80/auth/123").Return(nil, errors.New("error"))
+				mockCacher.EXPECT().Set("http://localhost:80/auth/123", gomock.Any(), time.Minute).Return(errors.New("error"))
```

Comment:
Validate the data that is being sent to this Set method instead of using gomock.Any(). The response is anyway from the mocked test handler, so this should be predictable.

Reply:
done